src/host/linalg.jl

```diff
@@ -127,6 +126,36 @@ function LinearAlgebra.copytri!(A::AbstractGPUMatrix{T}, uplo::AbstractChar, con
   A
 end
 
+## copy a triangular part of a matrix to another matrix
+
+if isdefined(LinearAlgebra, :copytrito!)
+    function LinearAlgebra.copytrito!(B::AbstractGPUMatrix, A::AbstractGPUMatrix, uplo::AbstractChar)
+        LinearAlgebra.BLAS.chkuplo(uplo)
+        m,n = size(A)
+        m1,n1 = size(B)
+        (m1 < m || n1 < n) && throw(DimensionMismatch("B of size ($m1,$n1) should have at least the same number of rows and columns than A of size ($m,$n)"))
+        if uplo == 'U'
+            gpu_call(A, B) do ctx, _A, _B
```

Comment:
This looks great, but I am tempted to ask you to write it with KernelAbstractions ;)

Reply:
We already have it :)
https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/main/src/block_krylov_utils.jl#L159-L171
What is the difference between what is generated by `gpu_call` and a kernel of `KernelAbstractions.jl`?

Reply:
`gpu_call` is the internal predecessor to KA.

The long-term plan is to move things to KA (https://github.com/JuliaGPU/GPUArrays.jl/pull/451) so this might be a good place to start.

Reply:
Moving everything at once is a lot of work, but I am happy to merge this as is and I would be ecstatic if you found time to take on linalg

Reply:
Ok, let's do that. :smiley: